health-services/plan-service/src/main/java/digit/service/validator/PlanEmployeeAssignmentValidator.java

```diff
@@ -205,11 +195,70 @@ private void validateCampaignDetails(String campaignId, String tenantId, PlanEmp
-        validateEmployeeJurisdiction(campaignResponse.getCampaignDetails().get(0), planEmployeeAssignment);
+        // Validate the provided jurisdiction for the plan employee assignment
+        validateEmployeeAssignmentJurisdiction(campaignResponse.getCampaignDetails().get(0), planEmployeeAssignment);
+
+        // Validates the jurisdiction assigned to Root role employee against MDMS
+        validateRootEmployeeJurisdiction(planEmployeeAssignment, mdmsData, campaignResponse.getCampaignDetails().get(0));
 
-        // Validates highest root jurisdiction for National roles against MDMS
-        validateNationalRole(planEmployeeAssignment, mdmsData, campaignResponse.getCampaignDetails().get(0));
+        // Validates the jurisdiction assigned to a non-Root employee against MDMS
+        validateEmployeeJurisdiction(planEmployeeAssignment, mdmsData, campaignResponse.getCampaignDetails().get(0));
+    }
+
+    /**
+     * This is a helper method to get the lowest or highest hierarchy for microplan from MDMS
+     *
+     * @param mdmsData        the mdms data
+     * @param lowestHierarchy if true, returns the lowest hierarchy for microplan
+     * @return returns the lowest or highest hierarchy for microplan
+     */
+    private String getMicroplanHierarchy(Object mdmsData, boolean lowestHierarchy) {
+
+        String jsonPathForMicroplanHierarchy = JSON_ROOT_PATH + MDMS_ADMIN_CONSOLE_MODULE_NAME + DOT_SEPARATOR + MDMS_MASTER_HIERARCHY_CONFIG + HIERARCHY_CONFIG_FOR_MICROPLAN;
+
+        List<Map<String, String>> hierarchyForMicroplan;
+
+        try {
+            log.info(jsonPathForMicroplanHierarchy);
+            hierarchyForMicroplan = JsonPath.read(mdmsData, jsonPathForMicroplanHierarchy);
+        } catch (Exception e) {
+            log.error(e.getMessage());
+            throw new CustomException(JSONPATH_ERROR_CODE, JSONPATH_ERROR_MESSAGE);
+        }
+
+        if (lowestHierarchy) {
+            return hierarchyForMicroplan.get(0).get(LOWEST_HIERARCHY_FIELD_FOR_MICROPLAN);
+        }
+
+        return hierarchyForMicroplan.get(0).get(HIGHEST_HIERARCHY_FIELD_FOR_MICROPLAN);
+    }
+
+    /**
+     * Validates that a non-Root role employee is not assigned to the highest or lowest hierarchy against MDMS
+     *
+     * @param planEmployeeAssignment The plan employee assignment provided in request
+     * @param mdmsData               mdms data from mdms v2
+     * @param campaignDetail         the campaign details for the corresponding campaign id
+     */
+    private void validateEmployeeJurisdiction(PlanEmployeeAssignment planEmployeeAssignment, Object mdmsData, CampaignDetail campaignDetail) {
+        if (!planEmployeeAssignment.getRole().contains(ROOT_ROLE)) {
+            List<String> jurisdiction = planEmployeeAssignment.getJurisdiction();
+
+            // Fetch the highest and lowest hierarchy for Microplan from MDMS
+            String lowestHierarchy = getMicroplanHierarchy(mdmsData, Boolean.TRUE);
```

Comment:
make changes here accordingly
keep the heirarchys in a map and use map.get(LOWEST_HEIRARCHY) in places where you need it